### PR TITLE
button to put cards down

### LINKED ISSUE #62: Frontend: Button to to put all cards "down"

### PR DESCRIPTION
**Pull request by ElenaGrafUZH**

### CODE REVIEW

**Review by EriCreator — CHANGES_REQUESTED** [review]
I'm sorry, but this is not the way to do this.
Fetching a possible move is ingame logic and should be handled by websockets in the first place.

The server should check for a possible move at the start of a players turn and send a (websocket) notification if there is no possible move. And this is when you set `movePossible`, which could also be part of our state (maybe right below `playerIndex`). 

The surrender cards endpoint isn't really necessary, but could be implemented if you really want it. Otherwise it's also possible to just randomly highlight one of the available cards.

**Comment by ElenaGrafUZH** [comment]
> I'm sorry, but this is not the way to do this. Fetching a possible move is ingame logic and should be handled by websockets in the first place.
> 
> The server should check for a possible move at the start of a players turn and send a (websocket) notification if there is no possible move. And this is when you set `movePossible`, which could also be part of our state (maybe right below `playerIndex`).
> 
> The surrender cards endpoint isn't really necessary, but could be implemented if you really want it. Otherwise it's also possible to just randomly highlight one of the available cards.

You can find the changes on the new commit. I hope it's better this way

File: src/components/views/BasicBoard.js [hunk 1]
```diff
@@ -166,6 +173,11 @@ const BasicBoard = (props) => {
   const undoHover = () => {
     setClassName("board icons-text");
     setRuleTypeShow(false);
+
+  };
+  const discardHand = () => {
+    surrenderCards();
+    state.movePossible = true;
```
Comment (EriCreator) [inline src/components/views/BasicBoard.js:180]: did you test this? this should not work without calling `setState(state);`

Comment (EriCreator) [reply src/components/views/BasicBoard.js:180]:
```suggestion
    state.movePossible = true;
    setState({ ...state });
```

**Review by EriCreator — APPROVED** [review]

**Pull request merged**